File: lösungen_lurch/PHP/matrix02.php
<?php

// Logische Lösung
function printMatrixRandomCharacter($number) {
  $digitsOfSquare = $number * $number;
  $randomNumber = rand(1, $digitsOfSquare);

  for ($i = 1; $i <= $digitsOfSquare; $i++) {
    if ($i == $randomNumber) {
      print "*";
    } else {
      print "#";
    }

    if ($i % $number == 0) {
      print "\n";
    }
  }
}

$userInput = (int)readline("Enter a number:");
printMatrixRandomCharacter($userInput);

// Mathematische Lösung
/*
public Vector3Int[] getGridCoordinatesForItem(Vector3Int pos, int item)
    {
        Vector2Int size = availableGridElements[item].size;
        Vector3Int[] allPositions = new Vector3Int[size.x * size.y];

        for (int i = 0; i < allPositions.Length; i++)
        {
            int x = i % size.x;
            int y = Mathf.FloorToInt(i / size.x);

            // 2D size VS on the grid VS world....
            allPositions[i] = new Vector3Int(pos.x + x, pos.y, pos.z - y);
        }

        return allPositions;
    }
*/

File: lösungen_lurch/Python/matrix02.py
import random

def print_matrix_random_character(number):
    digits_of_square = number * number
    random_number = random.randrange(1, digits_of_square)

    for i in range(1, digits_of_square+1):
        if i == random_number:
            print("*", end="")
        else:
            print("#", end="")

        if i % number == 0:
            print()


user_input = int(input("Enter a number:"))
print_matrix_random_character(user_input)
